File: algorithms/1603.design-parking-system/solution.rs
struct ParkingSystem {
    parking_spaces: Vec<i32>
}


/** 
 * `&self` means the method takes an immutable reference.
 * If you need a mutable reference, change it to `&mut self` instead.
 */
impl ParkingSystem {

    fn new(big: i32, medium: i32, small: i32) -> Self {
        ParkingSystem {
            parking_spaces: vec![0, big, medium, small]
        }
    }

    fn add_car(&mut self, car_type: i32) -> bool {
        if self.parking_spaces[car_type as usize] == 0 {
            return false;
        }

        self.parking_spaces[car_type as usize] -= 1;
        true
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * let obj = ParkingSystem::new(big, medium, small);
 * let ret_1: bool = obj.add_car(carType);
 */

File: algorithms/1603.design-parking-system/solution.rb
class ParkingSystem

=begin
  :type big: Integer
  :type medium: Integer
  :type small: Integer
=end
  def initialize(big, medium, small)
    @parking_spaces = [0, big, medium, small]
  end

=begin
  :type car_type: Integer
  :rtype: Boolean
=end
  def add_car(car_type)
    return false if @parking_spaces[car_type] == 0

    @parking_spaces[car_type] -= 1
    true
  end
end

# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem.new(big, medium, small)
# param_1 = obj.add_car(car_type)
